### DISCUSSION TITLE
How to upgrade timesketch instance installed with docker-compose ?

### DISCUSSION THREAD

**Discussion by presianbg** [google/timesketch#1999] [Q&A, quick solutions, support] [answered]
Hi,

I'm trying to update my 3 month old timesketch instance to the latest version following this guide:
https://github.com/google/timesketch/blob/master/docs/guides/admin/upgrade.md

But it looks like its not applicable for installations done via:
https://github.com/google/timesketch/blob/master/contrib/deploy_timesketch.sh

because trying to backup the postgresql results in:

```
pg_dump timesketch > backup-timesketch-db.sql
pg_dump: error: connection to database "timesketch" failed: FATAL:  role "postgres" does not exist
```

Also the `timesketch-web` instance do not have `git` installed. 

Is there a way to upgrade such installation at all ? 

Cheers,
PY

**Comment by jaegeral** [comment, marked as answer]
Good question, I do not have an answer available right now because the deploy timesketch script has been contributed to the project and is not really used by us

**Reply by presianbg** [reply]
I think i was able to update my instance and haven't lost my imported data.
What I did:

1. Skipped the database backup step, because I have VM backup;
2. No need of `git` in the container, because you we can clone the repo in any of those volumes from the host:
```
    volumes:
      - ./etc/timesketch:/etc/timesketch/
      - ./upload:/usr/share/timesketch/upload/
      - ./logs:/var/log/timesketch/
```
and proceed with the `root@<CONTAINER_ID>$ tsctl db current` checks.
It actually turned out that my db schema is the latest one... so no need of this step either.
3. Finally:
```
$ docker-compose pull
$ docker-compose down
$ docker-compose up -d
```

How can I check if I'm with the latest timesketch ?

Looking at the frontend and comparing with commit #1995 tells me that I succeeded to upgrade my setup:
![image](https://user-images.githubusercontent.com/2163486/142856079-73451db3-4fe6-4d34-9969-fc1b440fc571.png)

Cheers,
PY
